reference/5.1/Microsoft.PowerShell.Core/About/about_Functions_Advanced_Parameters.md

```diff
@@ -677,9 +693,9 @@ $Message = "bye"
 The **ValidateNotNull** attribute specifies that the parameter value can't be
 `$null`. PowerShell generates an error if the parameter value is `$null`.
 
-The **ValidateNotNull** attribute is designed to be used when the type of the
-parameter value isn't specified or when the specified type accepts a value of
-`$null`. If you specify a type that doesn't accept a `$null` value, such as a
+The **ValidateNotNull** attribute is designed to be used when the specified
+type accepts a value of `$null` or if the parameter is optional and the type is
+undefined. If you specify a type that doesn't accept a `$null` value, such as a
```

Comment:
See my earlier [comment](https://github.com/MicrosoftDocs/PowerShell-Docs/issues/6578#issuecomment-686586842) on this.